### DISCUSSION TITLE
Don't nest cloth config.

### DISCUSSION THREAD

**Discussion by K0-RR** [RakambdaOrg/FallingTree#199] [General]
This lib is highly popular and its size is huge - ~90% of the size of your mod. It gets included in many mods so the modpack with 80 mods is 30 MB bigger in size than needed.
Or maybe you can release an additional version without cloth?

**Comment by Rakambda** [comment]
I could do that yeah. It is already the case with forge where if cloth isn't there then the mod still loads fine.

However in the fabric version fabric the mod is way more linked with cloth so it's harder to dynamically link it. One example of this is that it loads the config file, as contrary to forge there's no conf loader in fabric by default.
The thing I don't really want is to force people to download cloth in order to have fallingtree working. I'd prefer have it "optional" and not "required".

If you really need it without the lib, you can always  compile it yourself and remove that line: https://github.com/RakSrinaNa/FallingTree/blob/4b0d6049e1c892d98cad2f55d11351634c988687/fabric/build.gradle.kts#L19
I'll see if I can automate the building of it.

If I can't, I'll keep the idea in mind but to be honest it seems quite a lot of effort for not that much. 30Mb more to download / have on a hard drive over a whole pack isn't really much nowadays.

**Reply by Rakambda** [reply]
This will be part of version 3.0.0, as the config format will change this will introduce a breaking change but the end jar file is indeed smaller (300-400kb instead of something like 1.2mb).

See #272
